src/io/linux.zig

```diff
@@ -564,6 +567,26 @@ pub const IO = struct {
                     };
                     completion.callback(completion.context, completion, &result);
                 },
+                .fsync => {
+                    const result: anyerror!void = blk: {
+                        if (completion.result < 0) {
+                            const err = switch (@as(posix.E, @enumFromInt(-completion.result))) {
+                                .INTR => {
+                                    completion.io.enqueue(completion);
+                                    return;
+                                },
+                                .BADF => error.FileDescriptorInvalid,
+                                .IO => error.InputOutput,
+                                .INVAL => unreachable,
+                                else => |errno| stdx.unexpected_errno("fsync", errno),
```

Comment:
I am not sure we've got all interesting error codes here, but we are going to crash on sync failure anyway, so this is fine

Reply:
Those are all the ones explicitly defined for `fsync`, but yeah it can fail with any of the error conditions defined for `read` or `write` too.